src/ICS20Transfer.sol

```diff
@@ -170,6 +134,49 @@ contract ICS20Transfer is IIBCApp, IICS20Transfer, IICS20Errors, Ownable, Reentr
         emit ICS20Timeout(packetData);
     }
 
+    /// @notice Create a MsgSendPacket for an ICS20 transfer
+    /// @notice This function is meant as a helper function to easily construct a correct MsgSendPacket
+    /// @param msg_ The message for sending a transfer
+    /// @return The constructed MsgSendPacket
+    function createMsgSendPacket(SendTransferMsg calldata msg_)
+        public
+        view
+        returns (IICS26RouterMsgs.MsgSendPacket memory)
+    {
+        require(msg_.amount > 0, ICS20InvalidAmount(msg_.amount));
+
+        // we expect the denom to be an erc20 address
+        address contractAddress = ICS20Lib.mustHexStringToAddress(msg_.denom);
+
+        string memory fullDenomPath;
+        try IBCERC20(contractAddress).fullDenomPath() returns (string memory ibcERC20FullDenomPath) {
```

Comment:
We should probably convert this to a static call. But there is another issue for static calls, so we don't need to address this here